frontend/device/android/device.lua

```diff
@@ -373,6 +372,20 @@ function Device:canExecuteScript(file)
     end
 end
 
+function Device:isValidPath(name)
+    local path, _ = util.splitFilePathName(name)
+    if android.isPathInsideSandbox(name) then
+        return true
+    else -- test if name is a symlink to the sandbox
+        path = path:gsub("/$", "") -- lua does not like trailing slash
+        if lfs.symlinkattributes(path, "mode") == "link" then
```

Comment:
since the symbolic link is always between "/sdcard" and the primary storage partition you can simplify the whole function to

```lua
function Device:isValidPath(path)
    if path:find("^/sdcard") ~= nil then
        path = path:gsub("/sdcard", android.getExternalStoragePath())
    end
    return android.isPathInsideSandbox(path)
end
```